### PR TITLE
Numerical types additions

### PR DESCRIPTION
**Pull request by fm-andres-rangel**

### CODE REVIEW

File: src/fmc/rational64.cpp [hunk 1]
```diff
@@ -0,0 +1,221 @@
...
+
+void fmc_rational64_abs(fmc_rational64_t *dest, const fmc_rational64_t *src) {
+  dest->num = std::abs(src->num);
+  dest->den = src->den;
+}
+
+void fmc_rational64_negate(fmc_rational64_t *dest,
+                           const fmc_rational64_t *src) {
+  dest->num = -src->num;
+  dest->den = -src->den;
```
Comment (fm-fede-ravchina) [inline src/fmc/rational64.cpp]: it should only negate one

File: src/fmc/rational64.cpp [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+  return (src->num == 1 || src->num == -1) && src->den == 0;
+}
+
+bool fmc_rational64_is_finite(const fmc_rational64_t *src) {
+  return (src->num != 1 && src->num != -1) && src->den != 0;
+}
+
+void fmc_rational64_abs(fmc_rational64_t *dest, const fmc_rational64_t *src) {
+  dest->num = std::abs(src->num);
+  dest->den = src->den;
```
Comment (fm-fede-ravchina) [inline src/fmc/rational64.cpp:206]: what if den is the negative one? or den is always positive?

File: src/fmc/rational64.cpp [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+}
+
+void fmc_rational64_negate(fmc_rational64_t *dest,
+                           const fmc_rational64_t *src) {
+  dest->num = -src->num;
+  dest->den = -src->den;
+}
+
+void fmc_rprice_dec(fmc_rational64_t *res, const fmc_rational64_t *src) {
```
Comment (fm-fede-ravchina) [inline src/fmc/rational64.cpp:215]: it should be fmc_rational64_dec

File: src/fmc/rational64.cpp [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+  fmc_rational64_sub(res, res, src);
+}
+
+void fmc_rprice_inc(fmc_rational64_t *res, const fmc_rational64_t *src) {
```
Comment (fm-fede-ravchina) [inline src/fmc/rational64.cpp:219]: same here

**Review by fm-fede-ravchina — DISMISSED** [review]

**Review by maxim-trokhimtchouk — APPROVED** [review]

File: include/fmc++/rational64.hpp [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+  static constexpr bool is_modulo = false;
+  static constexpr int radix = 2;
+  static constexpr int min_exponent = 0;
+  static constexpr int min_exponent10 = 0;
+  static constexpr int max_exponent = 0;
+  static constexpr int max_exponent10 = 0;
+  static constexpr bool traps = false;
+  static constexpr bool tinyness_before = true;
+  static constexpr fmc::rational64 min() noexcept {
+    return fmc::rational64(1, numeric_limits<int32_t>::max());
```
Comment (maxim-trokhimtchouk) [inline include/fmc++/rational64.hpp:228]: these need fixing

**Review by maxim-trokhimtchouk — CHANGES_REQUESTED** [review]

**Review by fm-fede-ravchina — APPROVED** [review]

**Review by maxim-trokhimtchouk — APPROVED** [review]

**Pull request merged**